code/__HELPERS/game.dm

```diff
@@ -213,6 +213,14 @@
 
 	return hear
 
+/proc/get_root_atom(atom/leaf)
+	var/atom/branch = leaf
+	while(branch.loc && !isturf(branch.loc))
+		branch = branch.loc
+	return branch
```

Comment:
We have `get_atom_on_turf()` already, does basically the same thing + an extra stop condition

Reply:
Thanks, I tried searching, but didn't find that.

Also, wow, that variable naming. It was throwing me for a loop, so I changed it.